specs/integration/search/CreateResourceWidget.js

```diff
@@ -0,0 +1,10 @@
+titleField: "input#title",
+openFileUploaderLink: '.resource-upload-dropzone input[type="file"]',
+
+fill: function fill(name) {
+    return  this.setTitleField(name)();
+},
+
+waitForUpload: function waitForUpload() {
+    setTimeout(function() {}, 7000); // TODO: fragile but improve your APM ;)
```

Comment:
This can't work, as it doesn't return a promise. What's probably happening is that you add a step that Watai retries for its own timeout.

Watai tries really hard to avoid any hard timeouts, which are the plague of integration testing, as they're unreliable and decrease performance. You should always have a way to work around this by using UI elements, with the side-effect that you can trace race conditions your users could encounter.

Here, for example, the best solution would be to replace this step with an intermediate assertion in the feature that would assert the absence of a loading indicator (`{ uploadSpinner: false }`, for example).

> If necessary, you could [adjust the timeout](https://github.com/MattiSG/Watai/wiki/Features#options) for this specific step.

If you don't have such a loading indicator, well, it might be worth considering adding it to the UI  :)  Making code testable usually improves its quality, the same usually goes for UI testing  :)

- - - - -

If really you can't find something your users have to understand when the step is done, I'd advise you to use a [custom function](https://github.com/MattiSG/Watai/wiki/Features#custom-function) straight in the feature (or in a data helper file) to wait for that time.